Resolved #2350 - Add ability to age out a redis lock (#2356)
nautobot/nautobot: nautobot/docs/configuration/optional-settings.md

---

## AVAILABLE_PREFIX_TIMEOUT

> Defaul: `600`

Reviewer: ```suggestion
Default: `600`
```

Environment Variable: `NAUTOBOT_AVAILABLE_PREFIX_TIMEOUT`

> Prevents a deadlock scenario where a Redis lock is achieved upon performing API to `/api/ipam/prefixes/{id}/available-prefixes/`. Default is set to 600 seconds (10 minutes) to be longer than any theorhetical API call time. This is to prevent a deadlock scenario where the server did not gracefully exit the with block when acquiring the Redis lock.

Reviewer: ```suggestion
Prevents a deadlock scenario where a Redis lock is achieved upon performing API to `/api/ipam/prefixes/{id}/available-prefixes/`. Default is set to 600 seconds (10 minutes) to be longer than any theoretical API call time. This is to prevent a deadlock scenario where the server did not gracefully exit the with block when acquiring the Redis lock.
```

> Prevents a deadlock scenario where a Redis lock is achieved upon performing API to `/api/ipam/prefixes/{id}/available-prefixes/`. Default is set to 600 seconds (10 minutes) to be longer than any theoretical API call time. This is to prevent a deadlock scenario where the server did not gracefully exit the with block when acquiring the Redis lock.

Reviewer: Perhaps:

```suggestion
Maximum duration of a Redis lock created when calling `/api/ipam/prefixes/{id}/available-prefixes/` or `/api/ipam/prefixes/{id}/available-ips/` to avoid inadvertently allocating the same prefix or IP to multiple simultaneous callers. Default is set to 600 seconds (10 minutes) to be longer than any theoretical API call time. This is to prevent a deadlock scenario where the server did not gracefully exit the `with` block when acquiring the Redis lock.
```

Author: Done
